File: backend/ruby_foundations/exercises/small_problems_101/easy_5/letter_counter_2.rb
def word_sizes(sentence)
  stat = Hash.new(0)
  sentence.split(' ').each do |word|
    word.delete(['^a-zA-Z'])
    stat[word.size] += 1
  end
  stat
end

puts word_sizes('Four score and seven.') == { 3 => 1, 4 => 1, 5 => 2 }
puts word_sizes('Hey diddle diddle, the cat and the fiddle!') == { 3 => 5, 6 => 3 }
puts word_sizes("What's up doc?") == { 5 => 1, 2 => 1, 3 => 1 }
puts word_sizes('') == {}

File: frontend/javascript/exercises/easy_3/letter_counter_2.js
function wordSizes(str) {
  var words = str.replace(/[^a-z\s]/gi, '').split(/\s+/);
  var wordCount = {};
  words.forEach(function (word) {
    var length = word.length.toString();
    if (length === '0') {
      return wordCount
    } else if (Object.keys(wordCount).indexOf(length) >= 0) {
      wordCount[length] += 1;
    } else {
      wordCount[length] = 1;
    }
  });
  return wordCount;
}

console.log(wordSizes('Four score and seven.'));                       
console.log(wordSizes('Hey diddle diddle, the cat and the fiddle!'));  
console.log(wordSizes('What\'s up doc?'));                             
console.log(wordSizes(''));                                          
console.log(wordSizes('Four   score  and      seven.'));